src/AdaptiveHoughGpuKernel.cpp

```diff
@@ -95,6 +122,78 @@ namespace HelixSolver
+            double yLineAtEnd    = inverse_r * INVERSE_A * (xEnd - phi);
+
+            const double y_for_x_min_acc = inverse_r * INVERSE_A * (PHI_BEGIN - phi);
+            const double y_for_x_max_acc = inverse_r * INVERSE_A * (PHI_END - phi);
+
+            if (y_for_x_min_acc <= Q_OVER_PT_BEGIN && y_for_x_max_acc >= Q_OVER_PT_END){
+
+                if (yLineAtBegin <= yEnd && section.yBegin <= yLineAtEnd)
+                {
+                    section.indices[counter] = index;
+                    counter++;
+                }
+            } else if (y_for_x_min_acc > Q_OVER_PT_BEGIN && y_for_x_min_acc < Q_OVER_PT_END){
+
+                if (yLineAtBegin <= yEnd && section.yBegin <= yLineAtEnd)
+                {
+                    section.indices[counter] = index;
+                    counter++;
+                }
+
+                yLineAtBegin  = inverse_r * INVERSE_A * (section.xBegin - phi) - 2 * M_PI * inverse_r * INVERSE_A;
+                yLineAtEnd    = inverse_r * INVERSE_A * (xEnd - phi) - 2 * M_PI * inverse_r * INVERSE_A;
+
+                if (yLineAtBegin <= yEnd && section.yBegin <= yLineAtEnd)
+                {
+                    section.indices[counter] = index;
+                    counter++;
+                }
+            } else if (y_for_x_max_acc < Q_OVER_PT_END && y_for_x_max_acc > Q_OVER_PT_BEGIN) {
+
+                if (yLineAtBegin <= yEnd && section.yBegin <= yLineAtEnd)
+                {
+                    section.indices[counter] = index;
+                    counter++;
+                }
+
+                yLineAtBegin  = inverse_r * INVERSE_A * (section.xBegin - phi) + 2 * M_PI * inverse_r * INVERSE_A;
+                yLineAtEnd    = inverse_r * INVERSE_A * (xEnd - phi) + 2 * M_PI * inverse_r * INVERSE_A;
+
+                if (yLineAtBegin <= yEnd && section.yBegin <= yLineAtEnd)
+                {
+                    section.indices[counter] = index;
+                    counter++;
+                }
+            }
+        }
+        section.counts =  counter + 1 == section.OUT_OF_RANGE_COUNTS ? 0 : counter; // setting this counter to 0 == indices are invalid
+        return counter;
+    }
+
+
+        uint8_t AdaptiveHoughGpuKernel::countHits_checkOrder(AccumulatorSection &section) const
+    {
+        const double xEnd = section.xBegin + section.xSize;
+        const double yEnd = section.yBegin + section.ySize;
+        uint16_t counter=0;
+        CDEBUG(DISPLAY_BOX_POSITION, section.xBegin<<","<<section.yBegin<<","<<xEnd<<","<<yEnd<<","<<section.divisionLevel<<":BoxPosition");
+
+        // here we can improve by knowing over which Points to iterate (i.e. indices of measurements), this is related to geometry
+        // this can be stored in section object maybe???
+
+        const uint32_t maxIndex = rs.size();
+        // 1st value will be coordinate of intersection point, the other is r of a given line
+        std::vector<std::pair<float, uint16_t>> cell_intersection_before;
```

Comment:
We can't use std::vectors. I.e. on GPU they do not work. This will need to be reworked with constant size std::array.
Good for now though.